crates/aptos/src/governance/mod.rs

```diff
@@ -83,53 +87,31 @@ impl CliCommand<Transaction> for SubmitProposal {
     }
 
     async fn execute(mut self) -> CliTypedResult<Transaction> {
-        // Validate the proposal metadata
-        let client = reqwest::ClientBuilder::default()
-            .tls_built_in_root_certs(true)
-            .build()
-            .map_err(|err| {
-                CliError::UnexpectedError(format!("Failed to build HTTP client {}", err))
-            })?;
-        let bytes = client
-            .get(self.metadata_url.clone())
-            .send()
-            .await
-            .map_err(|err| {
-                CliError::CommandArgumentError(format!(
-                    "Failed to fetch metadata url {}: {}",
-                    self.metadata_url, err
-                ))
-            })?
-            .bytes()
-            .await
-            .map_err(|err| {
-                CliError::CommandArgumentError(format!(
-                    "Failed to fetch metadata url {}: {}",
-                    self.metadata_url, err
-                ))
-            })?;
-        let metadata: ProposalMetadata = serde_json::from_slice(&bytes).map_err(|err| {
-            CliError::CommandArgumentError(format!(
-                "Metadata is not in a proper JSON format: {}",
-                err
-            ))
-        })?;
-        Url::parse(&metadata.source_code_url).map_err(|err| {
-            CliError::CommandArgumentError(format!(
-                "Source code URL {} is invalid {}",
-                metadata.source_code_url, err
-            ))
-        })?;
-        Url::parse(&metadata.discussion_url).map_err(|err| {
-            CliError::CommandArgumentError(format!(
-                "Discussion URL {} is invalid {}",
-                metadata.discussion_url, err
-            ))
-        })?;
+        // Check script file
+        let script_path = self.script_path.as_path();
+        if self.script_path.exists() {
```

Comment:
I feel like we do these 2 checks enough that they should be refactored into a util function or something

Reply:
Yeah, I'll just package them up now